tools/adventure-pack/goodies/typescript/Iterator.prototype.forEach/index.test.ts

```diff
@@ -0,0 +1,73 @@
+import { describe, expect, it, jest } from "@jest/globals";
+
+import "./index";
+
+describe("Iterator.prototype.forEach", () => {
+  it("calls the callback function for each array element", () => {
+    const array = [2, 3, 5, 7];
+    const callback = jest.fn();
+
+    array.values().forEach(callback);
+
+    expect(callback).toHaveBeenCalledTimes(array.length);
+  });
+
+  it("handles an empty iterator", () => {
+    const callback = jest.fn();
+
+    [].values().forEach(callback);
+    expect(callback).not.toHaveBeenCalled();
+  });
+
+  it("should execute side effects correctly", () => {
+    const array = [2, 3, 5, 7];
+    let sum = 0;
+
+    array.values().forEach((element) => {
+      sum += element;
+    });
+
+    expect(sum).toBe(17);
+  });
+
+  it("calls the callback for each map entry", () => {
+    const map = new Map([
+      ["car", "red"],
+      ["bus", "yellow"],
+      ["plane", "white"],
+    ]);
+    const callback = jest.fn();
+
+    map.entries().forEach(callback);
+    expect(callback).toHaveBeenCalledTimes(map.size);
+    expect(callback.mock.calls).toEqual([
+      [["car", "red"], 0],
+      [["bus", "yellow"], 1],
+      [["plane", "white"], 2],
+    ]);
+  });
+
+  it("calls the callback for each element in a set", () => {
+    const set = new Set([2, 3, 3, 4, 5, 5, 6]);
+    const callback = jest.fn();
+
+    set.values().forEach(callback);
+    expect(callback).toHaveBeenCalledTimes(set.size);
+    expect(callback.mock.calls).toEqual([
+      [2, 0],
+      [3, 1],
+      [4, 2],
+      [5, 3],
+      [6, 4],
+    ]);
+  });
+
+  it("should not fail if the callback returns a value", () => {
+    const array = [2, 3, 5, 7];
+    const callback = jest.fn(() => "ignored");
```

Comment:
If not using any of the Jest features, it probably doesn't need to be wrapped in `jest.fn`.

```suggestion
    const callback = () => "ignored";
```

Reply:
ack!